src/pages/ContentScripts/features/repo-collection/CollectionModal/CollectionEditor.tsx

```diff
@@ -0,0 +1,338 @@
+      )
+    )
+      .then((repositoryDescriptions) => {
+        setDataSource(repositoryDescriptions);
+      })
+      .catch((error) => {
+        console.error('Error fetching repository descriptions:', error);
+      });
+  }, []);
+
+  const initialValues = {
+    collectionName: isEdit ? collectionName : '', // 设置字段的初始值
+  };
+  const modalTitle = isEdit ? 'Collection Editor' : 'Creat a new collection';
+
+  const onSelectChange = (
+    newSelectedRowKeys: React.Key[],
+    selectedRows: DataType[]
+  ) => {
+    console.log('selectedRowKeys changed: ', newSelectedRowKeys);
+    console.log('selected Rows', selectedRows);
+    setNewRepoData(selectedRows.map((item) => item.name));
+    setSelectedRowKeys(newSelectedRowKeys);
+  };
+
+  const defaultSelectedRowKeys: React.Key[] = Array.from(
+    { length: collectionData.length },
+    (_, index) => index.toString()
+  );
+  const rowSelection = {
+    defaultSelectedRowKeys,
+    onChange: onSelectChange,
+  };
+
+  const handleInquireClick = () => {
+    // 在点击 "inquire" 按钮时，获取 Quick import 输入框的值
+    const inputValue = form.getFieldValue('Quick import');
+    async function fetchRepositories() {
+      try {
+        const result = await getUserOrOrgRepos(inputValue, isOrg);
+        // 在这里可以访问 "repositories"，它将包含获取的仓库信息
+        console.log('Repositories:', result);
+        let nextKey: number;
+        if (dataSource) {
+          nextKey = dataSource.length + 1;
+        } else {
+          nextKey = 1;
+        }
+        const addKeyValue = [
+          ...result.map((repo) => ({
+            key: (nextKey++).toString(),
+            name: repo.name,
+            description: repo.description,
+          })),
+        ];
+        if (dataSource) {
+          setDataSource([...dataSource, ...addKeyValue]);
+        } else {
+          setDataSource(addKeyValue);
+        }
+      } catch (error) {
+        // 处理错误
+        console.error('Error:', error);
+      }
+    }
+
+    fetchRepositories();
+  };
+
+  function handleImportClick() {
+    console.log('newRepoData', newRepoData);
+  }
+
+  return (
+    <Modal
+      width={1000}
+      open={open}
+      title={modalTitle}
+      okText="Confirm"
+      cancelText="Cancel"
+      onCancel={onCancel}
+      onOk={() => {
+        form
+          .validateFields()
+          .then((values) => {
+            form.resetFields();
+            onCreate(values, newRepoData);
+          })
+          .catch((info) => {
+            console.log('Validate Failed:', info);
+          });
+      }}
+    >
+      <Form
+        form={form}
+        layout="inline"
+        name="form_in_modal"
+        initialValues={initialValues}
+      >
+        <Form.Item
+          name="collectionName"
+          label="Name"
+          rules={[
+            {
+              required: true,
+              message: 'Please input the title of collection!',
+            },
+          ]}
+        >
+          <Input placeholder={'input name here'} />
```

Comment:
We need validate if the input name exists or not before we click on ok button. Duplicate collection names will cause serious bug. You can try replicating this by adding two collections with same name.

<img width="1306" alt="image" src="https://github.com/hypertrons/hypertrons-crx/assets/39271899/0551aad1-6c41-489d-a05f-ffafd636f332">

Reply:
This is a very important reminder! I added a judgment rule that will prompt the user when duplicate names appear.

<img width="1442" alt="image" src="https://github.com/hypertrons/hypertrons-crx/assets/50283262/59b4880e-372e-41ec-a0aa-0f9744a96e28">